src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource/config-detail.jelly

```diff
@@ -2,27 +2,48 @@
 <j:jelly xmlns:j="jelly:core" xmlns:c="/lib/credentials"
          xmlns:f="/lib/form"
          xmlns:scm="/jenkins/scm/api/form">
-  <j:choose>
-    <j:when test="${descriptor.apiUriSelectable}">
-    <f:entry title="${%API endpoint}" field="apiUri">
-      <f:select/>
-    </f:entry>
-    </j:when>
-    <j:otherwise>
-      <f:invisibleEntry>
-        <f:select field="apiUri"/>
-      </f:invisibleEntry>
-    </j:otherwise>
-  </j:choose>
   <f:entry title="${%Credentials}" field="credentialsId">
     <c:select checkMethod="post"/>
   </f:entry>
-  <f:entry title="${%Owner}" field="repoOwner">
-    <f:textbox/>
-  </f:entry>
-  <f:entry title="${%Repository}" field="repository">
-    <f:select/>
-  </f:entry>
+  <f:invisibleEntry>
+    <f:textbox name="isConfiguredByUrlDynamicValue" value="true"/>
+  </f:invisibleEntry>
+  <f:radioBlock name="isConfiguredByUrl" value="true" checked="true" title="${%Repository HTTPS URL}" inline="true">
```

Comment:
Like I mentioned earlier, I would rename this to `configuredByUrlRadioButton` so you can simplify the hidden field name to just `configuredByUrl`.

Also, maybe worth a comment in the Jelly that this is always true because we want to always show this configuration method by default, even for old instances, and that's why `getRepositoryUrl` always computes a return value.